react/ProductBreadcrumb.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { FC } from 'react'
+
+import { BreadcrumbList } from './modules/breadcrumb'
+import { getBaseUrl } from './modules/baseUrl'
+
+interface CategoryTreeItem {
+  name: string
+  href: string
+}
+
+export const productBreadcrumb = (
+  categoryTree?: CategoryTreeItem[],
+  productName?: string,
+  productSlug?: string
+): BreadcrumbList | {} => {
+  if (!categoryTree || !Array.isArray(categoryTree)) {
+    return {}
+  }
+
+  const baseUrl = getBaseUrl()
+
+  const categoryItems = categoryTree.map((category, index) => ({
+    '@type': 'ListItem',
+    position: index + 1,
+    name: category.name,
+    item: baseUrl + category.href,
+  }))
+
+  const productItem =
+    productName && productSlug
+      ? [
+          {
+            '@type': 'ListItem',
+            position: categoryItems.length + 1,
+            name: productName,
+            item: `${baseUrl}/${productSlug}/p`,
+          },
+        ]
+      : []
+
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'BreadcrumbList',
+    itemListElement: categoryItems.concat(...productItem),
+  }
```

Comment:
what do you think of:

```suggestion
if (productName && productSlug) {
  categoryItems.push({
    '@type': 'ListItem',
    position: categoryItems.length + 1,
    name: productName,
    item: `${baseUrl}/${productSlug}/p`,
  })
}

return {
  '@context': 'https://schema.org',
  '@type': 'BreadcrumbList',
  itemListElement: categoryItems,
}
```

Reply:
uhh much better 👌